Sources/SwiftProtobuf/JSONDecoder.swift

```diff
@@ -45,7 +52,10 @@ internal struct JSONDecoder: Decoder {
     if fieldCount > 0 {
       try scanner.skipRequiredComma()
     }
-    if let fieldNumber = try scanner.nextFieldNumber(names: fieldNameMap!) {
+    let fieldNumber = try scanner.nextFieldNumber(names: fieldNameMap!,
+                                                  messageType: messageType,
+                                                  extensionMap: extensions)
```

Comment:
Rather than passing the extensionMap ever call, should we pass it in the initializer?  it can't change during the life of the scanner, right?

Reply:
Good idea!  Same argument applies to `options`, so I've moved both of those down into the scanner.  That provoked several minor code simplifications regarding these values as well.